library/entity/status_effect/src/main/java/org/quiltmc/qsl/entity/effect/mixin/LivingEntityMixin.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity.effect.mixin;
+
+import java.util.Map;
+
+import org.jetbrains.annotations.NotNull;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.attribute.AttributeContainer;
+import net.minecraft.entity.effect.StatusEffect;
+import net.minecraft.entity.effect.StatusEffectInstance;
+
+import org.quiltmc.qsl.entity.effect.api.QuiltLivingEntityStatusEffectExtensions;
+import org.quiltmc.qsl.entity.effect.api.StatusEffectRemovalReason;
+
+@Mixin(LivingEntity.class)
+public abstract class LivingEntityMixin extends Entity implements QuiltLivingEntityStatusEffectExtensions {
+	@SuppressWarnings("ConstantConditions")
+	public LivingEntityMixin() {
+		super(null, null);
+		throw new IllegalStateException("Mixin ctor called??");
```

Comment:
Hmm, is this necessary?

Reply:
javac throws a fit if it can't find a constructor that calls the super constructor, so yes.
This is a thing with _all_ mixins that have to extend a superclass.

Reply:
No no no, I mean the exception; Yeah, the `super` call is indeed a Mixin thing, but something tells me the exception isn't necessary

Reply:
I think we already have mixins in QSL that do this? Let me check.

Reply:
Hm, apparently not. Will remove the exception.